### PR TITLE
Nexus 23683 helm ui upload

### PR DESCRIPTION
**Pull request by aivasenko**
Ticket: https://issues.sonatype.org/browse/NEXUS-23683

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package org.sonatype.repository.helm;
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java:1]: Missed header

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.sonatype.nexus.repository.security.VariableResolverAdapter;
+import org.sonatype.nexus.repository.upload.ComponentUpload;
+import org.sonatype.nexus.repository.upload.UploadDefinition;
+import org.sonatype.nexus.repository.upload.UploadHandlerSupport;
+import org.sonatype.nexus.repository.upload.UploadResponse;
+import org.sonatype.repository.helm.internal.util.HelmAttributeParser;
+
+import static org.sonatype.repository.helm.internal.HelmFormat.NAME;
+
+public class HelmUploadHandlerSupport
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java:18]: Sinse missed

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmUploadHandler.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import static org.sonatype.repository.helm.internal.HelmFormat.NAME;
+
+/**
+ * Support helm upload for web page
+ *
+ * @author yinlongfei
+ */
+@Singleton
+@Named(NAME)
+public class HelmUploadHandler
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmUploadHandler.java:52]: Since missed

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java [hunk 4]
```diff
@@ -83,6 +85,11 @@ public Content putIndex(final String path, final Content content, final AssetKin
     }
   }
 
+  @Override
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java:88]: final for attribute name
I am preferring to name it payload against content

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacet.java [hunk 5]
```diff
@@ -31,7 +33,14 @@
 {
   Content get(String path);
 
-  void upload(String path, Payload payload, final AssetKind assetKind) throws IOException;
+  Content upload(String path, Payload payload, final AssetKind assetKind) throws IOException;
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacet.java:36]: final missing

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/hosted/HelmHostedFacet.java [hunk 6]
```diff
@@ -17,9 +17,9 @@
 import org.sonatype.nexus.repository.Facet;
 import org.sonatype.nexus.repository.Facet.Exposed;
 import org.sonatype.nexus.repository.storage.Asset;
-import org.sonatype.nexus.repository.storage.TempBlob;
 import org.sonatype.nexus.repository.view.Content;
 import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.payloads.TempBlob;
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/hosted/HelmHostedFacet.java:22]: Orient should work with storage.TempBlob

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java [hunk 7]
```diff
@@ -58,7 +60,23 @@ public void upload(
...
+
+  @Override
+  public Content upload(
+      final String path,
+      TempBlob tempBlob,
+      HelmAttributes helmAttributes,
+      final Payload payload,
+      final AssetKind assetKind) throws IOException
+  {
+    // TODO: Write policy selector
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacetImpl.java:74]: Is is still required?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    this.contentPermissionChecker = contentPermissionChecker;
+    this.variableResolverAdapter = variableResolverAdapter;
+    this.helmPackageParser = helmPackageParser;
+  }
+
+  @Override
+  public UploadResponse handle(
+      final Repository repository, final ComponentUpload upload) throws IOException
+  {
+    return null;
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java:45]: strange function

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java [hunk 9]
```diff
@@ -52,9 +53,10 @@
   private final IndexYamlAbsoluteUrlRewriter indexYamlAbsoluteUrlRewriter;
 
   @Inject
-  public HelmContentFacetImpl(@Named(HelmFormat.NAME) final FormatStoreManager formatStoreManager,
-                              final HelmAttributeParser helmAttributeParser,
-                              final IndexYamlAbsoluteUrlRewriter indexYamlAbsoluteUrlRewriter)
+  public HelmContentFacetImpl(
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java:56]: Minor: redundant refactoring

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacet.java [hunk 10]
```diff
@@ -31,7 +33,14 @@
...
+
+  Content upload(
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacet.java:38]: missing final here too

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/hosted/HelmHostedFacetImpl.java [hunk 11]
```diff
@@ -25,16 +25,16 @@
 import org.sonatype.nexus.repository.storage.Asset;
 import org.sonatype.nexus.repository.storage.StorageFacet;
 import org.sonatype.nexus.repository.storage.StorageTx;
-import org.sonatype.nexus.repository.storage.TempBlob;
 import org.sonatype.nexus.repository.transaction.TransactionalDeleteBlob;
 import org.sonatype.nexus.repository.transaction.TransactionalStoreBlob;
 import org.sonatype.nexus.repository.transaction.TransactionalTouchBlob;
 import org.sonatype.nexus.repository.view.Content;
 import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.payloads.TempBlob;
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/orient/hosted/HelmHostedFacetImpl.java:33]: As @igorudovika mentioned above: "Orient should work with storage.TempBlob"

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacet.java (hunk 5 above)
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedFacet.java:36]: Why did you change the return type here? Looks like you don't use this method

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java (hunk 9 above)
Comment (aivasenko) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java:61]: Suggested by IDE. is it ok if we keep so?

Comment (chernikovmaksym) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java:61]: It's a minor comment so it's up to you. I'd suggest using "VCS only" in the refactoring menu

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java [hunk 12]
```diff
@@ -99,6 +101,11 @@ public Content putIndex(final String path, final Content content, final AssetKin
     }
   }
 
+  @Override
+  public TempBlob getTempBlob(Payload payload) {
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmContentFacetImpl.java:105]: You may want to make the method parameter `final`

**Comment by Darkvar** [comment]
+0 since I'm not familiar with upload functionality

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmUploadHandler.java [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+      final ContentPermissionChecker contentPermissionChecker,
+      final HelmAttributeParser helmPackageParser,
+      @Named("simple") final VariableResolverAdapter variableResolverAdapter,
+      final Set<UploadDefinitionExtension> uploadDefinitionExtensions)
+  {
+    super(contentPermissionChecker, helmPackageParser, variableResolverAdapter, uploadDefinitionExtensions);
+  }
+
+  @Override
+  public UploadResponse handle(Repository repository, ComponentUpload upload) throws IOException {
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmUploadHandler.java:65]: Final

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+import org.sonatype.nexus.repository.security.ContentPermissionChecker;
+import org.sonatype.nexus.repository.security.VariableResolverAdapter;
+import org.sonatype.nexus.repository.upload.UploadDefinition;
+import org.sonatype.nexus.repository.upload.UploadHandlerSupport;
+import org.sonatype.repository.helm.internal.util.HelmAttributeParser;
+
+import static org.sonatype.repository.helm.internal.HelmFormat.NAME;
+
+/**
+ * @since 1.0.11
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java:27]: Is this the discussed release version or it should be `1.next `here?

Comment (chernikovmaksym) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java:27]: Same for some other files in this PR

Comment (joedragons) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandlerSupport.java:27]: I have not seen this strategy used for community tho I've seen this asked a lot=)
I think there's no automation for community so it's a manual set.  Would be interested to learn otherwise and if it's project specific.
